frame/assets/src/lib.rs

```diff
@@ -153,6 +157,15 @@ pub trait Config: frame_system::Config {
 
 	/// Weight information for extrinsics in this pallet.
 	type WeightInfo: WeightInfo;
+
+	/// The flag for determining asset freezing and thawing support.
```

Comment:
I will prefer a feature flag enum instead of multiple booleans.

Reply:
@shawntabrizi wdyt? Also these small changes, I'm mostly impartial to. If you want to fix please do so, I'm really just trying to get some extensible token functionality for other pallets. The faster we can merge this the better :).

Reply:
since this is meant to be going into the main codebase, i'd prefer to keep code quality high. a single flag enum is the "right" way of doing it; three separate booleans leads to boilerplatey clutter all over the codebase.

Reply:
Should the enum support all combinations? W/o all combinations, something like?
```
enum AllowableFunctionalityFlag {
    Freezable,
    Mintable,
    Burnable,
    All,
    None,
}
```